### ISSUE TITLE
`php flarum migrate` crashes in-between versions

### ISSUE THREAD

**Issue by tobyzerner** [flarum/core#1056]
Originally reported here: https://discuss.flarum.org/d/4011-upgrade-to-beta-6

On a beta 5 installation, after you run `composer update` to pull the beta 6 code, immediately running `php flarum migrate` produces the following error:

> PHP Fatal error:  Uncaught exception 'Illuminate\Contracts\Container\BindingResolutionException' with message 'Unresolvable dependency resolving [Parameter # 1 [ <required> callable $userResolver ]] in class Flarum\Core\Access\Gate' in /vendor/illuminate/container/Container.php:828

**Comment by Luceos** [comment]
I'm not quite sure whether I understand the issue at hand. What is the solution we should implement? Shouldn't `php flarum migrate` run all mgirations of Flarum already? Or is the container error thrown because the updated code is trying to resolve an extension?

**Comment by franzliedke** [comment]
I saw this when trying to run `php flarum` locally, too. My solution was to remove the config.php file (so that Flarum thinks it is not installed) - I had wanted to reinstall anyway.

Doesn't quite explain the underlying issue, though. And not sure why you got this when updating... mine works again now, after reinstalling ;)

**Comment by rhyseli** [comment]
I've just got this error from a fresh install. I was not able to run all of the migrations and ran into a 'Could not find table `settings`' error.

I noticed this is because the table `config` should have been renamed to `settings` based on the migrations. I manually changed the table name.

Strangely, only some of the migrations were run before the application started to try and bootstrap itself, resulting in a bunch of errors like this. It seems to stop after the 'add_hide_to_discussions' migration.
